contracts/rootchain/RootChain.sol

```diff
@@ -3,220 +3,315 @@ pragma solidity ^0.4.24;
+
+        // exit successfully challenged
+        balances[msg.sender] = balances[msg.sender].add(minExitBond);
+        totalWithdrawBalance = totalWithdrawBalance.add(minExitBond);
+        
+        depositExits[nonce].state = 2;
+        emit ChallengedDepositExit(nonce, exit_.owner, exit_.amount);
     }
 
-    /// @param txPos [0] Plasma block number in which the challenger's transaction occured
-    /// @param txPos [1] Transaction Index within the block
-    /// @param txPos [2] Output Index within the transaction (either 0 or 1)
-    /// @param newTxPos  Same as the above but the pos of the uxto created by the spend tx
-    function challengeExit(uint256[3] txPos, uint256[3] newTxPos, bytes txBytes, bytes proof, bytes sigs, bytes confirmationSig)
+    // @param txPos            position of the invalid exiting transaction [blkNum, txIndex, outputIndex]
+    // @param newTxPos         position of the challenging transaction [blkNum, txIndex, outputIndex]
+    // @param txBytes          raw transaction bytes of the challenging transaction
+    // @param sigs             signatures of the inputs for this transaction
+    // @param proof            proof of inclusion for this merkle hash
+    // @param confirmSignature signature used to invalidate the invalid exit. Signature is over (merkleHash, block header)
+    function challengeTransactionExit(uint256[3] exitingTxPos, uint256[3] challengingTxPos, bytes txBytes, bytes sigs, bytes proof, bytes confirmSignature)
         public
     {
-        // txBytes verification
         RLPReader.RLPItem[] memory txList = txBytes.toRlpItem().toList();
-        require(txList.length == 15, "incorrect tx list");
-
-        // start-exit verification
-        uint256 priority = 1000000000*txPos[0] + 10000*txPos[1] + txPos[2];
-        uint256[3] memory utxoPos = exits[priority].utxoPos;
-        require(utxoPos[0] == txList[0 + 5 * newTxPos[2]].toUint(), "incorrect blocknum");
-        require(utxoPos[1] == txList[1 + 5 * newTxPos[2]].toUint(), "incorrect tx index");
-        require(utxoPos[2] == txList[2 + 5 * newTxPos[2]].toUint(), "incorrect output index");
+        require(txList.length == 17, "incorrect tx list");
 
-        /*
-           Confirmation sig:
-              txHash, sigs, block header
-          */
+        // invalid transcation should have a pending exit
+        uint256 priority = blockIndexFactor*exitingTxPos[0] + txIndexFactor*exitingTxPos[1] + exitingTxPos[2];
+        exit memory exit_ = txExits[priority];
+        require(exit_.state == 1, "no pending exit to challenge");
 
-        bytes32 txHash = keccak256(txBytes);
-        bytes32 merkleHash = keccak256(abi.encodePacked(txHash, sigs));
-        bytes32 root = childChain[newTxPos[0]].root;
-        bytes32 confirmationHash = keccak256(abi.encodePacked(txHash, sigs, root));
+        // ensure matching inputs
+        ensureMatchingInputs(exitingTxPos, txList);
```

Comment:
A double spend challenge wouldn't pass this function right? In a double spend tx, the inputs should at least partially match to that of the exiting transaction. For example, lets say exiting tx spends (1, 0, 0, 0) and creates (3, 0, 0, 0). But someone is able to challenge with a tx that spends (1, 0, 0, 0) but whose position is (2, 0, 0, 0). Then the input of the challenging tx != the position of the exiting transaction, but it is a valid challenge.

Reply:
if a tx is spending (1,0,0,0), the challenging transaction should always have (1,0,0,0) as one of it's inputs right?

Reply:
Not if you are proving that the exiting transaction was later spent. You can challenge with a spend of a tx (in example above some spend of (3, 0, 0, 0) or a double sign (example above). One challenge has inputs (3, 0, 0, 0) while the other has (1, 0, 0, 0)

Reply:
Oh i see, you talking about the case of the challenge where the confirm signature is also included in the inputs. In that case I should just check to see whether or not the confirm signature passed is matches at least one of the inputs confirm signature. 

Nice catch.